src/v/archival/scrubber.cc

```diff
@@ -0,0 +1,308 @@
+      bucket, manifest_path, manifest, manifest_rtc);
+
+    if (manifest_get_result == download_result::notfound) {
+        vlog(
+          archival_log.debug,
+          "Partition manifest get {} not found",
+          manifest_path);
+        result.status = purge_status::permanent_failure;
+        co_return result;
+    } else if (manifest_get_result != download_result::success) {
+        vlog(
+          archival_log.debug,
+          "Partition manifest get {} failed: {}",
+          manifest_path,
+          manifest_get_result);
+        result.status = purge_status::retryable_failure;
+        co_return result;
+    }
+
+    auto partition_r = co_await cloud_storage::remote_partition::erase(
+      _api, bucket, std::move(manifest), _as);
+
+    if (partition_r != cloud_storage::remote_partition::erase_result::erased) {
+        vlog(
+          archival_log.debug,
+          "One or more objects deletions failed for {}",
+          ntp);
+        result.status = purge_status::retryable_failure;
+        co_return result;
+    }
+
+    // Erase the partition manifest
+    vlog(archival_log.debug, "Erasing partition manifest {}", manifest_path);
+    retry_chain_node manifest_delete_rtc(
+      ss::lowres_clock::now() + 5s, 1s, &parent_rtc);
+    result.ops += 1;
+    auto manifest_delete_result = co_await _api.delete_object(
+      bucket,
+      cloud_storage_clients::object_key(manifest_path),
+      manifest_delete_rtc);
+    if (manifest_delete_result != upload_result::success) {
+        result.status = purge_status::retryable_failure;
+        co_return result;
+    }
+
+    vlog(
+      archival_log.info,
+      "Finished erasing partition {} from object storage in {} requests",
+      ntp,
+      result.ops);
+    co_return result;
+}
+
+scrubber::global_position scrubber::get_global_position() {
+    const auto& nodes = _members_table.local().nodes();
+    auto self = config::node().node_id();
+
+    // members_table doesn't store nodes in sorted container, so
+    // we compose a sorted order first.
+    auto node_ids = _members_table.local().node_ids();
+    std::sort(node_ids.begin(), node_ids.end());
+
+    uint32_t result = 0;
+    uint32_t total = 0;
+
+    // Iterate over node IDs earlier than ours, sum their core counts
+    for (auto i : node_ids) {
+        const auto cores = nodes.at(i).broker.properties().cores;
+        if (i < self) {
+            result += cores;
+        }
+        total += cores;
+    }
+
+    result += ss::this_shard_id();
+
+    return global_position{.self = result, .total = total};
+}
+
+ss::future<housekeeping_job::run_result>
+scrubber::run(retry_chain_node& parent_rtc, run_quota_t quota) {
+    auto gate_holder = _gate.hold();
+
+    run_result result{
+      .status = run_status::skipped,
+      .consumed = run_quota_t(0),
+      .remaining = quota,
+    };
+
+    if (!_enabled) {
+        co_return result;
+    }
+
+    const auto& markers = _topic_table.get_lifecycle_markers();
```

Comment:
What happens when the topic table updates its list of lifecycle markers potentially leading to iterator invalidation whilst we're looping? I think it would be reasonable to make a copy here. I generally wouldn't expect the number of markers to be large. If we're concerned about large allocations, we could just copy a chunk here.

Reply:
Good catch, fixed.